src/textual/geometry.py

```diff
@@ -127,6 +140,9 @@ def __contains__(self, other: Any) -> bool:
         width, height = self
         return width > x >= 0 and height > y >= 0
 
+    
+R = TypeVar("R", bound="Region")
```

Comment:
Is the typevar here? Is it to type `cls` in `from_corners`? I think Mypy can deduce the type without it.

Reply:
I added the `R` TypeVar to provide better return-value type hints for `from_corners` and `from_origin`.  In my opinion, it's a bit of an anti-pattern to hard-code the return-type in `classmethod`s that act as alternative constructor-methods. If you ever decide to subclass `Region`, the runtime implementation will still be perfectly valid, but the type hint will break.

For example, in [this gist](https://mypy-play.net/?mypy=latest&python=3.10&flags=show-error-codes%2Cstrict&gist=73e1d45899d7356dfe02578925295d0a), MyPy correctly infers that the type of `region_with_type_var_subclass_instance` is `RegionWithTypeVarSubclass`, but incorrectly assumes that the type of `region_no_type_var_subclass_instance` is `RegionNoTypeVar` (it is in fact `RegionNoTypeVarSubclass`).

I'm obviously happy to revert it if you don't think it's necessary, however!